docs/developers_guide/deploying_spack.rst

```diff
@@ -0,0 +1,454 @@
+using ``--spack``.
+
+On many machines, the ``/tmp`` directory is not a safe place to build spack
+packages.  Use ``--tmpdir`` to point to another place.
+
+The ``--recreate`` flag may not be strictly necessary but it’s a good idea.
+This will make sure both the bootstrapping conda environment (the one that
+installs mache to identify the machine) and the compass conda environment are
+created fresh.
+
+The ``--compiler`` flag is a list of one or more compilers to build for and the
+``--mpi`` flag is the corresponding list of MPI libraries.  To see what is
+supported on each machine, take a look at :ref:`dev_supported_machines`.
+
+Be aware that not all compilers and MPI libraries support Albany and PETSc, as
+discussed below.
+
+During the spack build, you may see:
+
+.. code-block:: none
+
+    ==> Error: no such environment 'dev_compass_1_2_0-alpha_5_intel_openmpi'
+    creating new environment: dev_compass_1_2_0-alpha_5_intel_openmpi
+
+That’s okay, this isn’t a real error and should hopefully be hidden soon.
+
+Testing spack with Albany
+-------------------------
+
+If you also want to build Albany, use the ``--with_albany`` flag.  Currently,
+this only works with Gnu compilers.  There is a file,
+`albany_support.txt <https://github.com/MPAS-Dev/compass/blob/main/conda/albany_supported.txt>`_,
+that lists supported compilers and MPI libraries on each machine.
+
+Here is an example:
+
+.. code-block:: bash
+
+    export TMPDIR=/lcrc/group/e3sm/${USER}/spack_temp
+    ./conda/configure_compass_env.py \
+        --conda ${CONDA_BASE} \
+        --mache_fork xylar/mache \
+        --mache_branch update_cime_machines \
+        --update_spack \
+        --spack /lcrc/group/e3sm/${USER}/spack_test \
+        --tmpdir ${TMPDIR} \
+        --compiler gnu \
+        --mpi openmpi \
+        --with_albany \
+        --recreate
+
+Testing spack with PETSc (and Netlib LAPACK)
+--------------------------------------------
+
+If you want to build PETSc (and Netlib LAPACK), use the ``--with_petsc`` and
+``--with_netlib_lapack`` flags.  Currently, this only works with some
+compilers, but that may be more that I was trying to limit the amount of work
+for the compass support team.  There is a file,
+`petsc_supported.txt <https://github.com/MPAS-Dev/compass/blob/main/conda/petsc_supported.txt>`_,
+that lists supported compilers and MPI libraries on each machine.
+
+Here is an example:
+
+.. code-block:: bash
+
+    export TMPDIR=/lcrc/group/e3sm/${USER}/spack_temp
+    ./conda/configure_compass_env.py \
+        --conda ${CONDA_BASE} \
+        --mache_fork xylar/mache \
+        --mache_branch update_cime_machines \
+        --update_spack \
+        --spack /lcrc/group/e3sm/${USER}/spack_test \
+        --tmpdir ${TMPDIR} \
+        --compiler intel gnu \
+        --mpi openmpi openmpi \
+        --with_netlib_lapack \
+        --with_petsc \
+        --recreate
+
+Testing compass
```

Comment:
```suggestion

Troubleshooting spack
------------------------------

If you encounter an error like:
```
==>   spack env activate dev_compass_1_2_0-alpha_4_gnu_mpich
==> Error: Package 'armpl' not found.
You may need to run 'spack clean -m'.
```
during the attempt to build spack, you will first need to find the path to `setup-env.sh` (see 
`compass/build_*/build*.sh`) and source that script to get the `spack` command, e.g.:

.. code-block:: bash

    source ${PSCRATCH}/spack_test/spack_for_mache_1.12.0/share/spack/setup-env.sh

Then run the suggested command:

.. code-block:: bash

    spack clean -m

After that, re-running `./conda/configure_compass_env.py` should work correctly.

This issue seems to be related to switching between spack v0.18 and v0.19 (used by different versions of compass).

Testing compass
```